clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1786,6 +1786,9 @@ void CodeGenFunction::GenerateCode(GlobalDecl GD, llvm::Function *Fn,
     if (SyclOptReport.HasOptReportInfo(FD)) {
       llvm::OptimizationRemarkEmitter ORE(Fn);
       for (auto ORI : llvm::enumerate(SyclOptReport.GetInfo(FD))) {
+        // Temporarily apply arg location to ensure SourceLocToDebugLoc
+        // picks up the expected file.
+        ApplyDebugLocation tmp_raii(*this, ORI.value().KernelArgLoc);
```

Comment:
Please use Pascal Case for naming variables. `TempRAII  is an option.

It also isn't clear to me where this is used.

Reply:
I've changed the variable name, hope it's ok. 

This variable isn't (directly) used by the subsequent code in this function - it's just a RAII variable to temporarily set the argument location to maintain the current behavior of this [non-llvm-upstream lit test](https://github.com/intel/llvm/blob/sycl/clang/test/SemaSYCL/kernel-arg-opt-report.cpp). I have updated the PR description with more details. Please let me know if more clarification is needed. Thanks.